File: Ontology/PropertyRestriction.cs
﻿namespace Ontology
{
    public abstract class PropertyRestriction: IPropertyRestriction
    {
        private readonly IPropertyExpression _propertyExpression;

        protected PropertyRestriction(
            IPropertyExpression propertyExpression
            )
        {
            _propertyExpression = propertyExpression;
        }

        IPropertyExpression IPropertyRestriction.PropertyExpression => _propertyExpression;

        public abstract void Accept(IClassExpressionVisitor visitor);

        public abstract bool Evaluate(
            IClassMembershipEvaluator evaluator,
            object                    individual);
    }
}

File: Web/ClientApp/src/app/Ontology/PropertyRestriction.ts
import { ClassExpression } from "./ClassExpression";
import { IClassExpressionVisitor } from "./IClassExpressionVisitor";
import { IPropertyExpression } from "./IPropertyExpression";
import { IPropertyRestriction } from "./IPropertyRestriction";

export class PropertyRestriction
    extends ClassExpression
    implements IPropertyRestriction
{
    protected constructor(
        public PropertyExpression: IPropertyExpression
        )
    {
        super();
    }

    Accept(
        visitor: IClassExpressionVisitor
        ): void
    {
        throw new Error("Method not implemented.");
    }
}
